crypto/ethsecp256k1/ethsecp256k1.go

```diff
@@ -28,54 +36,85 @@ const (
 // ----------------------------------------------------------------------------
 // secp256k1 Private Key
 
-var _ tmcrypto.PrivKey = PrivKey{}
-
-// PrivKey defines a type alias for an ecdsa.PrivateKey that implements
-// Tendermint's PrivateKey interface.
-type PrivKey []byte
+var (
+	_ cryptotypes.PrivKey  = &PrivKey{}
+	_ codec.AminoMarshaler = &PrivKey{}
+)
 
 // GenerateKey generates a new random private key. It returns an error upon
 // failure.
-func GenerateKey() (PrivKey, error) {
+func GenerateKey() (*PrivKey, error) {
 	priv, err := ethcrypto.GenerateKey()
 	if err != nil {
-		return PrivKey{}, err
+		return nil, err
 	}
 
-	return PrivKey(ethcrypto.FromECDSA(priv)), nil
+	return &PrivKey{
+		Key: ethcrypto.FromECDSA(priv),
+	}, nil
+}
+
+// Bytes returns the byte representation of the ECDSA Private Key.
+func (privKey PrivKey) Bytes() []byte {
+	return privKey.Key
 }
 
 // PubKey returns the ECDSA private key's public key.
-func (privkey PrivKey) PubKey() tmcrypto.PubKey {
-	ecdsaPKey := privkey.ToECDSA()
-	return PubKey(ethcrypto.CompressPubkey(&ecdsaPKey.PublicKey))
+func (privKey PrivKey) PubKey() cryptotypes.PubKey {
+	ecdsaPrivKey := privKey.ToECDSA()
+	return &PubKey{
+		Key: ethcrypto.CompressPubkey(&ecdsaPrivKey.PublicKey),
+	}
 }
 
-// Bytes returns the raw ECDSA private key bytes.
-func (privkey PrivKey) Bytes() []byte {
-	return CryptoCodec.MustMarshalBinaryBare(privkey)
+// Equals returns true if two ECDSA private keys are equal and false otherwise.
+func (privKey PrivKey) Equals(other cryptotypes.LedgerPrivKey) bool {
+	return privKey.Type() == other.Type() && subtle.ConstantTimeCompare(privKey.Bytes(), other.Bytes()) == 1
 }
 
-// Sign creates a recoverable ECDSA signature on the secp256k1 curve over the
-// Keccak256 hash of the provided message. The produced signature is 65 bytes
-// where the last byte contains the recovery ID.
-func (privkey PrivKey) Sign(msg []byte) ([]byte, error) {
-	return ethcrypto.Sign(ethcrypto.Keccak256Hash(msg).Bytes(), privkey.ToECDSA())
+// Type returns eth_secp256k1
+func (privKey PrivKey) Type() string {
+	return KeyType
 }
 
-// Equals returns true if two ECDSA private keys are equal and false otherwise.
-func (privkey PrivKey) Equals(other tmcrypto.PrivKey) bool {
-	if other, ok := other.(PrivKey); ok {
-		return bytes.Equal(privkey.Bytes(), other.Bytes())
+// MarshalAmino overrides Amino binary marshalling.
+func (privKey PrivKey) MarshalAmino() ([]byte, error) {
+	return privKey.Key, nil
+}
+
+// UnmarshalAmino overrides Amino binary marshalling.
+func (privKey *PrivKey) UnmarshalAmino(bz []byte) error {
```

Comment:
not sure what this does. is this supposed to unmrshal something? or is this setting the "amino encoded" public key to the bytes passed?